core/dynamo.c

```diff
@@ -978,7 +984,22 @@ dynamo_shared_exit(thread_record_t *toexit /* must ==cur thread for Linux */
      * client trying to use api routines that depend on fragment state.
      */
     instrument_exit();
-#endif
+# ifdef CLIENT_SIDELINE
+    /* We only need do a second synch-all if there are sideline client threads. */
+    synch_with_threads_at_exit(exit_synch_state(), false/*post-exit*/);
```

Comment:
And only on UNIX?  Or I guess we'll leave this code cross-platform for when the Windows terminate syscall is skipped.